classes/controller/AdminController.php

```diff
@@ -2542,7 +2542,7 @@ public function renderForm()
             $helper->tpl_vars = $this->getTemplateFormVars();
             $helper->show_cancel_button = (isset($this->show_form_cancel_button)) ? $this->show_form_cancel_button : ($this->display == 'add' || $this->display == 'edit');
 
-            $back = Tools::safeOutput(Tools::getValue('back', ''));
+            $back = urldecode(Tools::getValue('back', ''));
```

Comment:
This line of code (`$back = Tools::safeOutput(Tools::getValue('back', ''));`) can be found in many places (`AdminController`, `AdminCategoriesController`, `AdminFeaturesController`, etc.) . Have you tested the same behavior against these other places ?